pkgs/tools/backup/s3ql/default.nix

```diff
@@ -11,25 +11,45 @@ python3Packages.buildPythonApplication rec {
     sha256 = "1x0xj8clfs8fdczn8skc2wag5i4z47bsvlczn22iaf20hll1bb2w";
   };
 
-  checkInputs = [ which ] ++ (with python3Packages; [ cython pytest ]);
+  nativeBuildInputs = with python3Packages; [
+    cython
+  ];
+  buildInputs = [
+    sqlite
+  ];
+  checkInputs = with python3Packages; [
+    pytest
+  ] ++ [
+    which
+  ];
   propagatedBuildInputs = with python3Packages; [
-    sqlite apsw pycrypto requests defusedxml dugong llfuse
-    cython pytest pytest-catchlog google_auth google-auth-oauthlib
+    apsw
+    cryptography
+    defusedxml
+    dugong
+    google-auth-oauthlib
+    google_auth
+    llfuse
+    requests
+    systemd
+  ] ++ [
+    psmisc # `umount.s3ql` requires `fuser`
+    utillinuxMinimal.bin # `umount.s3ql` requires `umount`
   ];
 
   preBuild = ''
+    # http://www.rath.org/s3ql-docs/installation.html#installing-s3ql
+    rm ./MANIFEST.in # Build for release
     ${python3Packages.python.interpreter} ./setup.py build_cython build_ext --inplace
   '';
 
   checkPhase = ''
-    # Removing integration tests
-    rm tests/t{4,5,6}_*
-    pytest tests
+    HOME=/build pytest tests
```

Comment:
Not sure if /build is not an implementation detail, do not see it documented anywhere. Anyway, `$TMPDIR` might be a better choice.

Reply:
`HOME=/build` was a suggestion from sphalerite on NixOS IRC, to fix the `/homeless-shelter` issue: <https://logs.nix.samueldr.com/nixos/2019-08-26#;>. I'm not sure what exactly the suggestion was based on.

Reply:
It was based on the fact that the build is carried out in a writeable `/build` directory. But given that it is not documented anywhere and that it was changed in the past, I would consider it an implementation detail and use `HOME=$TMPDIR` instead.

Reply:
Oh no, why did I suggest that D: yes, I'm all for changing that.

Reply:
Done. Ironically `$TMPDIR` was `/build` when I tested it.